src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -129,7 +129,15 @@ public OAuthHandler(HttpClient backchannel)
                 properties.StoreTokens(authTokens);
             }
 
-            return AuthenticateResult.Success(await CreateTicketAsync(identity, properties, tokens));
+            try
+            {
+                var ticket = await CreateTicketAsync(identity, properties, tokens);
+                return AuthenticateResult.Success(ticket);
+            }
+            catch(Exception ex)
```

Comment:
If we wrap every invocation to `HandleRemoteAuthenticateAsync` with a try-catch, a conscious decision should be made to justify the two means of report errors (exception property vs. thrown exception).

This maybe helpful, but I'll say we need a case of unhandled exception in `HandleRemoteAuthenticateAsync` itself to study if such measure is helpful. Rather than get ahead of this issue itself.

I'll add comment to the signature of `HandleRemoteAuthenticateAsync` to state user should handle exception in it or face a leaked exception.